rewrite-javascript/rewrite/src/util.ts

```diff
@@ -14,5 +14,12 @@
  * limitations under the License.
  */
 export async function mapAsync<T, U>(arr: T[], fn: (t: T, i: number) => Promise<U | undefined>): Promise<U[]> {
-    return (await Promise.all(arr.map(fn))).filter(v => v !== undefined);
+    const results: U[] = [];
+    for await (const [i, item] of arr.entries()) {
+        const result = await fn(item, i);
+        if (result !== undefined) {
+            results.push(result);
+        }
+    }
```

Comment:
We don't have an async iterable here, so there is no need for `for async`.
```suggestion
    for (let i = 0; i < arr.length; i++) {
        const result = await fn(arr[i], i);
        if (result !== undefined) {
            results.push(result);
        }
    }
```